execution/batch.go

```diff
@@ -66,10 +66,19 @@ func (bsm *BatchSizeManager) ShouldTakeNextRecord() bool {
 	if bsm.curBatchSize >= bsm.batchSize {
 		return false
 	}
-	bsm.curBatchSize++
 	return true
 }
 
+// You can use this to process records in a way other than one by one.
+func (bsm *BatchSizeManager) RecordsLeftToTake() int {
+	return bsm.batchSize - bsm.curBatchSize
+}
+
+// You can use this to process records in a way other than one by one.
+func (bsm *BatchSizeManager) MarkRecordsProcessed(count int) {
+	bsm.curBatchSize += count
```

Comment:
Why is this stored in memory?

Reply:
1. It always has been 2. Because this value doesn't influence the correctness of the final output of the system. So there's no need for it to be transactional or whatever.